src/Sarif/Baseline/ResultMatching/SarifLogResultMatcher.cs

```diff
@@ -67,22 +67,25 @@ public IEnumerable<SarifLog> Match(IEnumerable<SarifLog> previousLogs, IEnumerab
                 {
                     baselineRuns = runsByToolPrevious[key];
                 }
-                IEnumerable<Run> currentRuns = new Run[0];
 
+                IEnumerable<Run> currentRuns = new Run[0];
                 if (runsByToolCurrent.ContainsKey(key))
                 {
                     currentRuns = runsByToolCurrent[key];
                 }
 
-                resultToolLogs.Add(BaselineSarifLogs(baselineRuns, currentRuns));
+                if (baselineRuns.Any() && currentRuns.Any())
```

Comment:
>baselineRuns.Any() && currentRuns.Any() [](http://example.com/codeflow?start=20&length=39)

If one of those values is empty, we will receive an exception.

To prevent that, let's skip tool names that does not match.